source/appModules/powerpnt.py

```diff
@@ -1128,6 +1128,36 @@ def _getFormatFieldAndOffsets(self, offset, formatConfig, calculateOffsets=True)
 			formatField["link"] = True
 		return formatField, (startOffset, endOffset)
 
+	def _setCaretOffset(self, offset: int):
+		if not 0 <= offset <= (maxLength := self._getStoryLength()):
```

Comment:
I think this might help with parsing a bit

```suggestion
		if not (0 <= offset <= (maxLength := self._getStoryLength())):
```